src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -225,15 +249,35 @@ else if (editMode == 'C') {
                 }
             }
         }
-
         this.comment = message.toString();
-
         int endOfFirstLine = this.comment.indexOf('\n');
         if (endOfFirstLine == -1) {
-            this.title = this.comment;
+            this.title = this.comment.trim();
         } else {
-            this.title = this.comment.substring(0, endOfFirstLine);
+            this.title = this.comment.substring(0, endOfFirstLine).trim();
+        }
+         if(!showEntireCommitSummaryInChanges){
+            this.title = splitString(this.title, TRUNCATE_LIMIT);
+        }
+    }
+
+    public static String splitString(String msg, int lineSize) {
+
+        StringBuilder sb = new StringBuilder(lineSize);
```

Comment:
Splitting the string on spaces seems like it will change the string by removing adjacent spaces.  That seems unnecessary.

We can get the same result with msg.lastIndexOf(' ', lineSize) can't we, without splitting and recombining the string.  Refer to the [prototype](https://github.com/MarkEWaite/git-plugin/blob/12e2f740aa8f3fffe1c3a48223bdedd806e37fa3/src/main/java/hudson/plugins/git/GitChangeSet.java#L286) I had crafted for an example.